api/base/exceptions.py

```diff
@@ -83,6 +83,10 @@ def json_api_exception_handler(exc, context):
     return response
 
 
+class NotImplementedError(APIException):
```

Comment:
This will shadow Python's built-in `NotImplementedError`. How about `EndpointNotImplementedError` or `HTTPNotImplementedError`?